backend/apis/clients/add-client.api.js

```diff
@@ -193,19 +203,69 @@ app.post("/api/clients", (req, res, next) => {
             registeredVoter,
             addedBy
           ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
+
+          INSERT INTO intakeData (
+            clientId,
+            dateOfIntake,
+            clientSource,
+            couldVolunteer,
+            addedBy
+          ) VALUES (?, ?, ?, ?, ?);
         `,
-          [...contactInfoValues, ...demographicsValues]
+          [...contactInfoValues, ...demographicsValues, ...intakeDataValues]
         );
 
-        connection.query(insertOther, (err, result, fields) => {
+        connection.query(insertOther, (err, results, fields) => {
           if (err) {
             connection.rollback();
             return databaseError(req, res, err, connection);
           }
 
-          connection.commit();
-          res.send({
-            success: true
+          const intakeDataResult = results[2];
+          const intakeDataId = intakeDataResult.insertId;
+
+          if (req.body.intakeServices.length === 0) {
+            connection.commit();
+            connection.release();
+
+            res.send({
+              success: true
+            });
+
+            return;
+          }
+
+          const intakeServicesValues = req.body.intakeServices.reduce(
```

Comment:
Me being curious, what is reduce doing?

Reply:
We want to transform an array of service ids into an array that can be used in the MySQL query as values. Each mysql query needs both the `intakeId` and the `serviceId`, so we need to turn each service id into two values in the array

```js
const intakeId = 423

// Starting value
[1, 2, 3]

// After we reduce
[423, 1, 423, 2, 423, 3]
```